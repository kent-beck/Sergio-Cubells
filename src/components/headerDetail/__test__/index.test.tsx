import { fireEvent, render, screen } from "@testing-library/react";
import HeaderDetail from "..";

const mockedUseNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...(jest.requireActual("react-router-dom") as any),
  useNavigate: () => mockedUseNavigate,
}));

describe("Render a ", () => {
  const title = "Nuevo titulo";

  test("headerDetail", () => {
    const tree = render(<HeaderDetail title="Nuevo Titulo" />).asFragment();
    expect(tree).toMatchSnapshot();
  });

  test("header title", () => {
    render(<HeaderDetail title={title} />).asFragment();

    const titleElement = screen.getByText(title);
    expect(titleElement).toBeInTheDocument();
  });

  test("alt image", () => {
    render(<HeaderDetail title={title} />);

    const testImage = document.querySelector("img") as HTMLImageElement;

    expect(testImage.alt).toContain("Back icon");
  });

  test("back icon src", () => {
    render(<HeaderDetail title={title} />);

    const testImage = document.querySelector("img") as HTMLImageElement;

    expect(testImage.src).toContain(
      "https://configurador.seat.es/seat-cc/assets/icons/unofficial/flat-arrow-left.svg"
    );
  });
});

describe("Click on back icon", () => {
  test("use navigate is called", () => {
    render(<HeaderDetail title="Nuevo Titulo" />);

    const backIcon = screen.getByTestId("backIcon");

    fireEvent.click(backIcon);

    expect(mockedUseNavigate).toBeCalled();
  });

  test("use navigate is called with /", () => {
    render(<HeaderDetail title="Nuevo Titulo" />);

    const backIcon = screen.getByTestId("backIcon");

    fireEvent.click(backIcon);

    expect(mockedUseNavigate).toBeCalledWith("/");
  });
});
