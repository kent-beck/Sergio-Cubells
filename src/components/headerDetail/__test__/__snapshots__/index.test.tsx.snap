// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a  headerDetail 1`] = `
<DocumentFragment>
  <div
    class="detailHeaderContainer"
  >
    <div
      class="detailTitleContainer"
    >
      <img
        alt="Back icon"
        class="backIcon"
        data-testid="backIcon"
        src="https://configurador.seat.es/seat-cc/assets/icons/unofficial/flat-arrow-left.svg"
      />
      <strong
        class="detailTitleCar"
      >
        Nuevo Titulo
      </strong>
    </div>
  </div>
</DocumentFragment>
`;
